### PR TITLE
Processor stats into separate struct

### PR DESCRIPTION
**Pull request by fracasula**
## Description of the change

* moving stats into own attribute (see `processor.HandleT`) to better understand processor dependencies
* typos and linting
* fix: honouring context during sleep

## Notion Link

> Notion Link

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by atzoum — APPROVED** [review]
LGTM

File: processor/manager.go [hunk 1]
```diff
@@ -61,7 +58,7 @@ func (proc *LifecycleManager) Stop() {
 }
 
 // New creates a new Processor instance
-func New(ctx context.Context, clearDb *bool, gwDb, rtDb, brtDb, errDb *jobsdb.HandleT, tenanatDB multitenant.MultiTenantI, reporting types.ReportingI) *LifecycleManager {
+func New(ctx context.Context, clearDb *bool, gwDb, rtDb, brtDb, errDb *jobsdb.HandleT, tenantDB multitenant.MultiTenantI, reporting types.ReportingI) *LifecycleManager {
```
[the remark this answers is not in the document]
Comment (fracasula) [reply processor/manager.go:61]: I am in favour of using options and make it a variadic function. Once I'm more familiar with the codebase next time I come across this bit I'll definitely try something :+1:

**Review by chrikar — APPROVED** [review]
Nice! 🙌🏼

**Comment by chrikar** [comment]
Looks like a test is failing, but I'm not if its related to this PR:
```
integration_test.go:261:
--
822 | Error Trace:    integration_test.go:261
823 | Error:          Condition never satisfied
824 | Test:           TestDynamicClusterManager
825 | controller.go:137: missing call(s) to *mock_backendconfig.MockBackendConfig.WaitForConfig(is anything) /codebuild/output/src051184173/src/github.com/rudderlabs/rudder-server/app/cluster/integration_test.go:206
826 | controller.go:137: missing call(s) to *mocks_transformer.MockTransformer.Setup() /codebuild/output/src051184173/src/github.com/rudderlabs/rudder-server/app/cluster/integration_test.go:207
827 | controller.go:137: aborting test due to missing call(s)
828 | --- FAIL: TestDynamicClusterManager (1.02s)
829 | FAIL
830 | coverage: 84.6% of statements
```

**Comment by fracasula** [comment]
> Looks like a test is failing, but I'm not if its related to this PR:
> 
> ```
> integration_test.go:261:
> --
> 822 | Error Trace:    integration_test.go:261
> 823 | Error:          Condition never satisfied
> 824 | Test:           TestDynamicClusterManager
> 825 | controller.go:137: missing call(s) to *mock_backendconfig.MockBackendConfig.WaitForConfig(is anything) /codebuild/output/src051184173/src/github.com/rudderlabs/rudder-server/app/cluster/integration_test.go:206
> 826 | controller.go:137: missing call(s) to *mocks_transformer.MockTransformer.Setup() /codebuild/output/src051184173/src/github.com/rudderlabs/rudder-server/app/cluster/integration_test.go:207
> 827 | controller.go:137: aborting test due to missing call(s)
> 828 | --- FAIL: TestDynamicClusterManager (1.02s)
> 829 | FAIL
> 830 | coverage: 84.6% of statements
> ```

I was asking in #ask-infra if I could retry the build because it looks like I don't have the permissions to do so but I'm looking into it thanks @chrikar 

https://rudderlabs.slack.com/archives/C02AASYS6R1/p1649663540597209

**Review by cisse21 — APPROVED** [review]

**Pull request merged**